src/backend/drm/output.rs

```diff
@@ -0,0 +1,459 @@
+use std::{
+    cell::RefCell,
+    collections::HashMap,
+    os::fd::AsFd,
+    sync::{Arc, RwLock},
+};
+
+use drm::control::{self, connector, crtc, Mode};
+use drm_fourcc::{DrmFormat, DrmFourcc, DrmModifier};
+
+use crate::{
+    backend::{
+        allocator::{
+            dmabuf::{AsDmabuf, Dmabuf},
+            gbm::GbmDevice,
+            Allocator,
+        },
+        renderer::{element::RenderElement, Bind, Color32F, DebugFlags, Renderer, Texture},
+    },
+    output::OutputModeSource,
+};
+
+use super::{
+    compositor::{
+        DrmCompositor, FrameError, FrameMode, FrameResult, RenderFrameError, RenderFrameErrorType,
+        RenderFrameResult,
+    },
+    exporter::ExportFramebuffer,
+    DrmDevice, DrmError, Planes,
+};
+
+type CompositorList<A, F, U, G> = Arc<RwLock<HashMap<crtc::Handle, RefCell<DrmCompositor<A, F, U, G>>>>>;
```

Comment:
Not primarily, but this was a nice side effect and is already used in use_mode.

The main reason was faking atomic access when changing the active outputs. So basically whenever a output is added or removed. The second was ownership.

We need to be able to access all active crtc while making sure we have unique access. Using Arc and cloning it around makes it hard to track aliveness of the output. We could store a Weak in the output manager, but tbh I really want to avoid that. Working with Weak is a pain sometimes. Also it would require locking all outputs before access, making this unnecessary complicated. The idea was that RwLock gives us that for free, concurrent read access and unique write access. Also it makes it easier to reason about who owns the outputs imo.